core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -84,6 +85,14 @@ case class SparkListenerBlockManagerRemoved(time: Long, blockManagerId: BlockMan
 @DeveloperApi
 case class SparkListenerUnpersistRDD(rddId: Int) extends SparkListenerEvent
 
+@DeveloperApi
+case class SparkListenerExecutorAdded(executorId: String, executorDetails: ExecutorDetails)
+  extends SparkListenerEvent
+
+@DeveloperApi
+case class SparkListenerExecutorRemoved(executorId: String, executorDetails: ExecutorDetails)
```

Comment:
This makes the event more self describing. Otherwise you'd need to also listen on the executorAdded event, have a hashmap then look it up in the executor removed method to find the details. 

That being said, I don't really feel that strongly so we can remove the executorDetails structure if you are concerned.